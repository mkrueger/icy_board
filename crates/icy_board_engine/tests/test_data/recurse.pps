DECLARE PROCEDURE FOO(VAR BYTE X)
BYTE C

FOO(C)
PRINTLN "END:",  C

PROCEDURE FOO(VAR BYTE X)
  BYTE C2
  C2 = X + 1
  if (X > 3) RETURN
  X = X * 2
  PRINTLN C2 ,":", X
  FOO(C2)
  PRINTLN C2 ,":", X
ENDPROC