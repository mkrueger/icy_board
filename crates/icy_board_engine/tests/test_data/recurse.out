1:0
2:2
3:4
4:6
4:6
6:4
4:2
2:0
END:0